cmd/paramgen/internal/paramgen.go

```diff
@@ -0,0 +1,628 @@
+		})
+		if found != nil {
+			// already found the type
+			break
+		}
+	}
+	if found == nil {
+		return nil, nil, fmt.Errorf("could not find type %v in package %v", typeName, pkg.Name)
+	}
+	return found, file, nil
+}
+
+func (p *paramsParser) findImportSpec(se *ast.SelectorExpr) (*ast.ImportSpec, error) {
+	impName := se.X.(*ast.Ident).Name
+	for _, i := range p.file.Imports {
+		if (i.Name != nil && i.Name.Name == impName) ||
+			strings.HasSuffix(strings.Trim(i.Path.Value, `"`), impName) {
+			return i, nil
+		}
+	}
+	return nil, fmt.Errorf("could not find import %q", impName)
+}
+
+func (p *paramsParser) attachPrefix(f *ast.Field, params map[string]sdk.Parameter) map[string]sdk.Parameter {
+	// attach prefix if a tag is present or if the field is named
+	prefix := p.getTag(f.Tag, TagParamName)
+	if prefix == "" && len(f.Names) > 0 {
+		prefix = p.formatFieldName(f.Names[0].Name)
+	}
+	// if it's a struct, then prefix is the struct name
+	if n, ok := f.Type.(*ast.Ident); ok && prefix == "" {
+		prefix = p.formatFieldName(n.Name)
+	}
+	if prefix == "" {
+		// no prefix to attach
+		return params
+	}
+
+	prefixedParams := make(map[string]sdk.Parameter)
+	for k, v := range params {
+		prefixedParams[prefix+"."+k] = v
+	}
+	return prefixedParams
+}
+
+func (p *paramsParser) isBuiltinType(name string) bool {
+	switch name {
+	case "string", "bool", "int", "uint", "int8", "uint8", "int16", "uint16", "int32", "uint32", "int64", "uint64",
+		"byte", "rune", "float32", "float64":
+		return true
+	default:
+		return false
+	}
+}
+
+func (p *paramsParser) parseSingleParameter(f *ast.Field, t sdk.ParameterType) (name string, param sdk.Parameter, err error) {
+	var fieldName string
+	if len(f.Names) == 1 {
+		fieldName = f.Names[0].Name
+	} else {
+		switch v := f.Type.(type) {
+		case *ast.Ident:
+			fieldName = v.Name
+		case *ast.SelectorExpr:
+			fieldName = v.Sel.Name
+		default:
+			return "", sdk.Parameter{}, fmt.Errorf("unexpected type: %T", f.Type)
+		}
+	}
+
+	name = p.getTag(f.Tag, TagParamName)
+	if name == "" {
+		// if there's no tag use the formatted field name
+		name = p.formatFieldName(fieldName)
+	}
+
+	var validations []sdk.Validation
+	validate := p.getTag(f.Tag, TagParamValidate)
+	if validate != "" {
+		validations, err = p.parseValidateTag(validate)
+		if err != nil {
+			return "", sdk.Parameter{}, err
+		}
+	}
+
+	// check if a required validation is in the list
+	isRequired := false
+	for _, v := range validations {
+		if _, ok := v.(sdk.ValidationRequired); ok {
+			isRequired = true
+		}
+	}
+
+	// replace field name with parameter name in description so that the user
+	// can write normal go docs referencing the field name
+	desc := strings.ReplaceAll(f.Doc.Text(), fieldName, name)
+	desc = strings.ReplaceAll(desc, "\n", " ")
+	desc = strings.Trim(desc, " ")
+
+	return name, sdk.Parameter{
+		Default:     p.getTag(f.Tag, TagParamDefault),
+		Description: desc,
+		Validations: validations,
+		Required:    isRequired,
```

Comment:
This is a deprecated field, we shouldn't set it. The information will already be included in `Validations`.

Reply:
It would yes, and I ignored it at first, but it seemed weird in the output file that all the required fields are false, so I added it.
But yes I agree.